### ISSUE TITLE
Access Violation Exception after upgrading from .NET 8 to .NET 9

### ISSUE THREAD

**Issue by karmeli87** [dotnet/runtime#110769]
### Description

We recently upgraded from .NET 8 to .NET 9 and start to experience crashes in our tests

```
Faulting application name: testhost.exe, version: 17.1200.24.56501, time stamp: 0x63740000
Faulting module name: coreclr.dll, version: 9.0.24.52809, time stamp: 0x672049fc
Exception code: 0xc0000005
Fault offset: 0x00000000002eea8b
Faulting process id: 0x2c8c
Faulting application start time: 0x01db4ea81cad46d8
Faulting application path: c:\Jenkins\workspace\Testsv70_Winx64\s\test\SlowTests\bin\Release\net9.0\testhost.exe
Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\9.0.0\coreclr.dll
Report Id: 3804b1ee-81a8-4f06-bef5-760530d90989
Faulting package full name: 
Faulting package-relative application ID: 
```

```
Application: testhost.exe
CoreCLR Version: 9.0.24.52809
.NET Version: 9.0.0
Description: The process was terminated due to an unhandled exception.
Stack:
   at System.Runtime.ExceptionServices.InternalCalls.RhpCallFinallyFunclet(Byte*, Void*, ExInfo*)
   at System.Runtime.ExceptionServices.InternalCalls.RhpCallFinallyFunclet(Byte*, Void*, ExInfo*)
   at System.Runtime.EH.InvokeSecondPass(ExInfo ByRef, UInt32, UInt32)
   at System.Runtime.EH.DispatchEx(System.Runtime.StackFrameIterator ByRef, ExInfo ByRef)
   at System.Runtime.EH.RhRethrow(ExInfo ByRef, ExInfo ByRef)
   at Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler`2[[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, 
Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, Culture=neutral, 
PublicKeyToken=7cec85d7bea7798e]].ReceiveReplicationBatches()
   at Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler`2[[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, 
Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=9.0.0.0, Culture=neutral, 
PublicKeyToken=7cec85d7bea7798e]].DoIncomingReplication()
   at Raven.Server.Utils.PoolOfThreads+PooledThread.DoWork()
   at Raven.Server.Utils.PoolOfThreads+PooledThread.Run()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
``` 

Will attach the dump privately.
https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326

### Reproduction Steps

Not at the moment.

### Expected behavior

not to crash with an AVE

### Actual behavior

crashing with an AVE

### Regression?

Yes, this was working without crashing on .NET 8

**Comment by janvorli** [comment]
It looks like a GC hole. Could you please share one more dump from a crash so that I can get more confidence on that? It would be great to get a dump with enabled stresslog which would allow me to see a recent history of GC stack walks / relocations. To do that, please set the following env variables:
```
DOTNET_StressLog=1 
DOTNET_LogLevel=7 
DOTNET_LogFacility=80103 
DOTNET_StressLogSize=2000000 
DOTNET_TotalStressLogSize=40000000
``` 

The stresslog is an in-memory log that gets included in the crash dump and can be dumped via the !stresslog command in WinDbg.

**Comment by karmeli87** [comment]
Hi,

Thanks for looking into that, we will set those env variables and wait for the next dump.

Since the upgrade to .NET 9, we get almost every test run on linux a seg fault, while on windows we get only once a day or two the above AVE.

If you like we can also share the linux dumps, but I don't want to mix several issues in one ticket and prefer to tackle one bug at a time.

But if you believe it can shed more light into current issue we can share those dumps as well.

**Comment by janvorli** [comment]
The Linux crash is a completely different thing. It looks like a JIT bug. Can you please create a separate issue for it? I've shared the call stack and crash reason in this gist: https://gist.github.com/janvorli/e9ce073e24f58ccf1629e3cfa0f6c414, so you can cut and paste it into the issue description.

**Comment by jkotas** [comment]
> https://gist.github.com/janvorli/e9ce073e24f58ccf1629e3cfa0f6c414, so you can cut and paste it into the issue description.

This looks like a duplicate of #109981

**Comment by karmeli87** [comment]
According to https://github.com/dotnet/runtime/issues/109981#issuecomment-2528071350 setting the following `DOTNET_JitEnableInductionVariableOpts=0` should workaround this issue.

We will try that and I report back once we will experience another crash.

**Comment by karmeli87** [comment]
I have shared another linux dump here https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10815576

with those env var set
```
DOTNET_JitEnableInductionVariableOpts=0

DOTNET_StressLog=1 
DOTNET_LogLevel=7 
DOTNET_LogFacility=80103 
DOTNET_StressLogSize=2000000 
DOTNET_TotalStressLogSize=40000000
```

Please let me know if should open a separate issue for those crashes.

**Comment by janvorli** [comment]
@karmeli87 the crash is exactly the same as the one before (linux).

**Comment by karmeli87** [comment]
Hi @janvorli thanks for not forgetting about that :-) 

Since it was crashing every single run on linux we decide to revert back to .Net 8 until the fix will be available in an official release (9.0.2)

Once it is available we will update back to the latest version and wait for the windows crash which is much more rare.

**Comment by janvorli** [comment]
@karmeli87 I was wondering if you have already updated to 9.0.2 and have some more details on the Windows crash?

**Comment by karmeli87** [comment]
Hi @janvorli, 
We didn't moved to 9.0.2 yet. 

But we do have a test branch on 9.0.2 that I didn't run for a while (it wasn't crashing for 5 consecutive runs at the time, which approved the fix for the linux case). 

Once I will get back to testing it, I will report back.

**Comment by karmeli87** [comment]
We got another crash 
```
Faulting application name: testhost.exe, version: 17.1300.25.10604, time stamp: 0x63740000
Faulting module name: coreclr.dll, version: 9.0.225.6610, time stamp: 0x6789558d
Exception code: 0xc0000005
Fault offset: 0x000000000004dd45
Faulting process id: 0x2690
Faulting application start time: 0x01db90c199cc28d6
Faulting application path: c:\Jenkins\workspace\Testsv71_Winx64_Debug\s\test\SlowTests\bin\Debug\net9.0\testhost.exe
Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\9.0.2\coreclr.dll
Report Id: ce9793bc-df10-4014-992a-26ddf62fffe5
```

I have attached the dump [here](https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10844587)

It run as Debug, but without other env variables.

This time the stacktrace is completely different and the heap seems to be very corrupted. 

```
0:062> k
 # Child-SP          RetAddr               Call Site
00 (Inline Function) --------`--------     coreclr!LoaderAllocator::IsCollectible [D:\a\_work\1\s\src\coreclr\vm\loaderallocator.hpp @ 575] 
01 000000bc`6a8fa1a0 00007ffa`95882025     coreclr!VirtualCallStubManager::ResolveWorker+0x4d [D:\a\_work\1\s\src\coreclr\vm\virtualcallstub.cpp @ 1486] 
02 000000bc`6a8fa360 00007ffa`958f44dd     coreclr!VSD_ResolveWorker+0x335 [D:\a\_work\1\s\src\coreclr\vm\virtualcallstub.cpp @ 1382] 
03 000000bc`6a8fa480 00007ffa`3dbd5fc5     coreclr!ResolveWorkerAsmStub+0x6d [D:\a\_work\1\s\src\coreclr\vm\amd64\VirtualCallStubAMD64.asm @ 36] 
04 000000bc`6a8fa530 00007ffa`9587c1fe     Raven_Server!Raven.Server.Documents.Indexes.Index.DoIndexingWork+0xbe5
05 000000bc`6a8fa580 00007ffa`3a4f6a89     coreclr!CallFinallyFunclet+0x11e [D:\a\_work\1\s\src\coreclr\vm\exceptionhandling.cpp @ 7997] 
06 000000bc`6a8fa630 00007ffa`3b46c700     0x00007ffa`3a4f6a89
07 000000bc`6a8fa760 00007ffa`3ee15388     0x00007ffa`3b46c700
08 000000bc`6a8fa880 00007ffa`958f2da3     0x00007ffa`3ee15388
09 000000bc`6a8fa8b0 00007ffa`95871418     coreclr!CallDescrWorkerInternal+0x83 [D:\a\_work\1\s\src\coreclr\vm\amd64\CallDescrWorkerAMD64.asm @ 74] 
0a 000000bc`6a8fa8f0 00007ffa`95862171     coreclr!DispatchCallSimple+0x60 [D:\a\_work\1\s\src\coreclr\vm\callhelpers.cpp @ 248] 
0b 000000bc`6a8fa980 00007ffa`95862261     coreclr!RethrowNew+0x101 [D:\a\_work\1\s\src\coreclr\vm\jithelpers.cpp @ 3161] 
0c 000000bc`6a8fc440 00007ffa`3dbd5e8e     coreclr!IL_Rethrow+0x81 [D:\a\_work\1\s\src\coreclr\vm\jithelpers.cpp @ 3177] 
0d 000000bc`6a8fc590 00007ffa`9587c3ff     Raven_Server!Raven.Server.Documents.Indexes.Index.DoIndexingWork+0xaae
0e 000000bc`6a8fc5e0 00007ffa`3b46c838     coreclr!CallCatchFunclet+0x17f [D:\a\_work\1\s\src\coreclr\vm\exceptionhandling.cpp @ 7751] 
0f 000000bc`6a8fc6f0 00007ffa`3b46c97d     0x00007ffa`3b46c838
10 000000bc`6a8fc810 00007ffa`958f2da3     0x00007ffa`3b46c97d
11 000000bc`6a8fc840 00007ffa`95871418     coreclr!CallDescrWorkerInternal+0x83 [D:\a\_work\1\s\src\coreclr\vm\amd64\CallDescrWorkerAMD64.asm @ 74] 
12 000000bc`6a8fc880 00007ffa`958619e1     coreclr!DispatchCallSimple+0x60 [D:\a\_work\1\s\src\coreclr\vm\callhelpers.cpp @ 248] 
13 000000bc`6a8fc910 00007ffa`95861874     coreclr!DispatchManagedException+0x169 [D:\a\_work\1\s\src\coreclr\vm\exceptionhandling.cpp @ 5698] 
14 000000bc`6a8fdfd0 00007ffa`95861843     coreclr!DispatchManagedException+0x24 [D:\a\_work\1\s\src\coreclr\vm\exceptionhandling.cpp @ 5715] 
15 000000bc`6a8fe4d0 00007ffa`958617ce     coreclr!ThrowNew+0x67 [D:\a\_work\1\s\src\coreclr\vm\jithelpers.cpp @ 3075] 
16 000000bc`6a8fe520 00007ffa`3b4d7b88     coreclr!IL_Throw+0x8e [D:\a\_work\1\s\src\coreclr\vm\jithelpers.cpp @ 3100] 
17 000000bc`6a8fe670 00007ffa`3a47146d     System_Private_CoreLib!System.Threading.CancellationToken.ThrowOperationCanceledException+0x38 [/_/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationToken.cs @ 368] 
18 000000bc`6a8fe6b0 00007ffa`3dbe0db1     System_Private_CoreLib!System.Threading.CancellationToken.ThrowIfCancellationRequested+0x1d [/_/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationToken.cs @ 362] 
19 000000bc`6a8fe6e0 00007ffa`3dbd57d7     Raven_Server!Raven.Server.Documents.Indexes.Workers.MapItems.Execute+0x6e1
1a 000000bc`6a8fec50 00007ffa`3dbcf1a2     Raven_Server!Raven.Server.Documents.Indexes.Index.DoIndexingWork+0x3f7
1b 000000bc`6a8fef40 00007ffa`3dbcca03     Raven_Server!Raven.Server.Documents.Indexes.Index.ExecuteIndexing+0xbb2
1c 000000bc`6a8ff750 00007ffa`39b6f0f5     Raven_Server!Raven.Server.Documents.Indexes.Index.<StartIndexingThread>b__167_0+0x103
1d 000000bc`6a8ff810 00007ffa`39b6d007     Raven_Server!Raven.Server.Utils.PoolOfThreads.PooledThread.DoWork+0x185
1e 000000bc`6a8ff940 00007ffa`388bbbc5     Raven_Server!Raven.Server.Utils.PoolOfThreads.PooledThread.Run+0xa7
1f 000000bc`6a8ff9a0 00007ffa`958f2da3     System_Private_CoreLib!System.Threading.ExecutionContext.RunInternal+0xe5 [/_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs @ 179] 
20 000000bc`6a8ffa10 00007ffa`95871418     coreclr!CallDescrWorkerInternal+0x83 [D:\a\_work\1\s\src\coreclr\vm\amd64\CallDescrWorkerAMD64.asm @ 74] 
21 000000bc`6a8ffa50 00007ffa`958e14b3     coreclr!DispatchCallSimple+0x60 [D:\a\_work\1\s\src\coreclr\vm\callhelpers.cpp @ 248] 
22 000000bc`6a8ffae0 00007ffa`9585ef35     coreclr!ThreadNative::KickOffThread_Worker+0x63 [D:\a\_work\1\s\src\coreclr\vm\comsynchronizable.cpp @ 158] 
23 (Inline Function) --------`--------     coreclr!ManagedThreadBase_DispatchInner+0xd [D:\a\_work\1\s\src\coreclr\vm\threads.cpp @ 7110] 
24 000000bc`6a8ffb40 00007ffa`9585ee5d     coreclr!ManagedThreadBase_DispatchMiddle+0x79 [D:\a\_work\1\s\src\coreclr\vm\threads.cpp @ 7154] 
25 000000bc`6a8ffbf0 00007ffa`9585efeb     coreclr!ManagedThreadBase_DispatchOuter+0x8d [D:\a\_work\1\s\src\coreclr\vm\threads.cpp @ 7313] 
26 (Inline Function) --------`--------     coreclr!ManagedThreadBase_FullTransition+0x28 [D:\a\_work\1\s\src\coreclr\vm\threads.cpp @ 7358] 
27 (Inline Function) --------`--------     coreclr!ManagedThreadBase::KickOff+0x28 [D:\a\_work\1\s\src\coreclr\vm\threads.cpp @ 7393] 
28 000000bc`6a8ffc60 00007ffa`b32c4cb0     coreclr!ThreadNative::KickOffThread+0x7b [D:\a\_work\1\s\src\coreclr\vm\comsynchronizable.cpp @ 230] 
29 000000bc`6a8ffcc0 00007ffa`b3afedcb     kernel32!BaseThreadInitThunk+0x10
2a 000000bc`6a8ffcf0 00000000`00000000     ntdll!RtlUserThreadStart+0x2b
```


```
0:062> !verifyheap
Segment          Object           Failure                          Reason
02dc0b17f378     029bfb23d908     FreeObjectReference              Object 29bfb23d908 contains free object at offset 10: 29c0f1620f0
02dc0b17f378     029bfb7f81f0     FreeObjectReference              Object 29bfb7f81f0 contains free object at offset 10: 29c1a789a38
02dc0b185930     029c20c1b4e0     SyncBlockMismatch                Object 29c20c1b4e0 System.WeakReference<Voron.Impl.Paging.Pager+State> had a SyncBlock index of 3695 but the runtime has no matching SyncBlock
02dc0b185930     029c20c8fd68     FreeObjectReference              Object 29c20c8fd68 contains free object at offset 18: 29c0f310378
02dc0b185930     029c20cd7650     FreeObjectReference              Object 29c20cd7650 contains free object at offset 18: 29c064c6f10
...
02dc0b18b9a8     029c458b29c8     InvalidObjectReference           Object 29c458b29c8 has a bad member at offset 20: 29c474f6800
02dc0b18bba0     029c4662ef10     InvalidObjectReference           Object 29c4662ef10 has a bad member at offset 38: 29c4747a900
02dc0b18bba0     029c467f82e8     FreeObjectReference              Object 29c467f82e8 contains free object at offset 10: 29c01e693b8

7,999,179 objects verified, 22,643 errors.
```

**Comment by karmeli87** [comment]
Hi @janvorli, so we are back into looking at .net 9 again and we updated to 9.0.5 version.

I ran with the env variables that you requested
```
DOTNET_StressLog=1 
DOTNET_LogLevel=7 
DOTNET_LogFacility=80103 
DOTNET_StressLogSize=2000000 
DOTNET_TotalStressLogSize=40000000
``` 

and we are having a crash with the following stack trace
```
(Inline Function) --------`--------     : --------`-------- --------`-------- --------`-------- --------`-------- : coreclr!LoaderAllocator::IsCollectible
00000049`d11faed0 00007ffe`b1f80d45     : 0000020a`a1029e50 00000049`d11fb0c0 00000049`d11fb260 0000000d`00000000 : coreclr!VirtualCallStubManager::ResolveWorker+0x4d
00000049`d11fb090 00007ffe`b1ff47ad     : 00000000`00000006 0000020b`15117040 00000049`d11ff6d0 00007ffe`55356d30 : coreclr!VSD_ResolveWorker+0x335
00000049`d11fb1b0 00007ffe`6146a688     : 0000020b`15117040 00000000`00000000 00007ffe`b22fb0d0 00000049`d11ff6d0 : coreclr!ResolveWorkerAsmStub+0x6d
00000049`d11fb260 00007ffe`b1f7af1e     : 00000049`d11ff6d0 00000049`d11fb4c0 00007ffe`55356d30 00000049`d11fc068 : Raven_Server!Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler<Raven.Server.ServerWide.Context.DocumentsContextPool,Raven.Server.ServerWide.Context.DocumentsOperationContext>.ReceiveReplicationBatches+0x1bc8
00000049`d11fb2f0 00007ffe`565ccd89     : 00007ffe`6146a603 00000000`00000006 00000000`0000000a 00000049`d11fb3c8 : coreclr!CallFinallyFunclet+0x11e
00000049`d11fb3a0 00007ffe`57b125f2     : 00000049`d11fbc10 00000049`d11fb578 00007ffe`61469016 00000049`d11fb590 : System_Private_CoreLib!System.Runtime.EH.InvokeSecondPass(ExInfo ByRef, UInt32, UInt32)+0xffffffff`be184fb9
00000049`d11fb4d0 00007ffe`58126fd8     : 00000049`d11fbcd0 00000049`d11fbc10 00007ffe`616bb6a8 00000049`d11fbe60 : System_Private_CoreLib!System.Runtime.EH.DispatchEx(System.Runtime.StackFrameIterator ByRef, ExInfo ByRef)+0xffffffff`bf6cb222
00000049`d11fb5f0 00007ffe`b1ff3073     : 00000049`d11fd7e0 00000049`d11fbc10 00000000`00000555 00000fff`cc28d491 : System_Private_CoreLib!System.Runtime.EH.RhRethrow(ExInfo ByRef, ExInfo ByRef)+0xffffffff`bfcdfc58
00000049`d11fb620 00007ffe`b1f70198     : 00000049`d11fbe60 00007ffe`6146a7a1 00000049`d11fbda0 00000049`d11fbda0 : coreclr!CallDescrWorkerInternal+0x83
00000049`d11fb660 00007ffe`b1f5e1d1     : 0000024b`6f849e00 00000049`d11fb7f0 00000049`d11fd7e0 0000024b`c52f0cf0 : coreclr!DispatchCallSimple+0x60
00000049`d11fb6f0 00007ffe`b1f5e2c1     : 00000049`d11fd570 0000020b`15155098 0000020b`12486f58 00007ffe`55356d30 : coreclr!RethrowNew+0x101
00000049`d11fd1b0 00007ffe`6146a5c0     : 00007ffe`55356d30 00007ffe`b1f7b327 00000049`d11fd7e0 00007ffe`6146a48e : coreclr!IL_Rethrow+0x81
00000049`d11fd300 00007ffe`b1f7b11f     : 00000049`d11fd570 00000049`d11fd5b0 00000049`d11fdc18 00000049`d11ff6d0 : Raven_Server!Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler<Raven.Server.ServerWide.Context.DocumentsContextPool,Raven.Server.ServerWide.Context.DocumentsOperationContext>.ReceiveReplicationBatches+0x1b00
00000049`d11fd390 00007ffe`57b12658     : 00000049`d11fd8a0 00007ffe`6146a48e 00000049`d11fd7e0 00000049`d11fd7e0 : coreclr!CallCatchFunclet+0x17f
00000049`d11fd4a0 00007ffe`57b1279d     : 00000049`d11fd8a0 00000049`d11fd7e0 00000049`d11ff930 00000000`00000000 : System_Private_CoreLib!System.Runtime.EH.DispatchEx(System.Runtime.StackFrameIterator ByRef, ExInfo ByRef)+0xffffffff`bf6cb288
00000049`d11fd5c0 00007ffe`b1ff3073     : 0000020b`12486f58 00000049`d11fd7e0 00007ffe`5525c6a8 00007ffe`b1f5d8f3 : System_Private_CoreLib!System.Runtime.EH.RhThrowEx(System.Object, ExInfo ByRef)+0xffffffff`bf6cb74d
00000049`d11fd5f0 00007ffe`b1f70198     : 00000049`d11fe4c0 0000020b`148e18f0 0000024b`6f849e00 00007ffe`b1f7b774 : coreclr!CallDescrWorkerInternal+0x83
00000049`d11fd630 00007ffe`b1f5da41     : 0000024b`6f849e00 00000049`d11fd7c0 0000020b`12486f58 0000024b`c52f0cf0 : coreclr!DispatchCallSimple+0x60
00000049`d11fd6c0 00007ffe`b1f5d8d4     : 00000000`00000000 0000020b`15117080 0000020b`12486f58 00000000`00000000 : coreclr!DispatchManagedException+0x169
00000049`d11fed80 00007ffe`b1f5d8a3     : 00000049`d11ff310 00007ffe`57be4e39 0000020a`a3c44478 0000024b`382cb0b0 : coreclr!DispatchManagedException+0x24
00000049`d11ff280 00007ffe`b1f5d82e     : 0000020b`12486f58 00007ffe`b1f8a29e 0000024b`382cb0b0 0000020a`a1c01c30 : coreclr!ThrowNew+0x67
00000049`d11ff2d0 00007ffe`57be7208     : 00000000`00000001 0000020b`12486f40 0000020b`1516f190 0000020b`12486f18 : coreclr!IL_Throw+0x8e
00000049`d11ff420 00007ffe`56faba28     : 0000020b`12486f18 00007ffe`b1fbedbf 00000049`d11ff4b8 00007ffe`b1f8aa17 : System_Private_CoreLib!System.Threading.CancellationToken.ThrowOperationCanceledException+0x38
00000049`d11ff460 00007ffe`57b1e280     : 0000020b`12486f18 00000000`ffffffff 0000020b`1516f190 0000020b`1516f190 : System_Private_CoreLib!System.Threading.ManualResetEventSlim.Wait+0x3a8
00000049`d11ff5a0 00007ffe`57b1df61     : 0000020b`12486ec0 00007ffe`546806db 00000000`00000000 0000020b`15192b40 : System_Private_CoreLib!System.Threading.Tasks.Task.SpinThenBlockingWait+0xf0
00000049`d11ff620 00007ffe`5753c88e     : 0000020b`1516f190 0000020b`15155718 0000020b`15190b60 0000020b`15190bb0 : System_Private_CoreLib!System.Threading.Tasks.Task.InternalWaitCore+0x71
00000049`d11ff6a0 00007ffe`61469016     : 0000020b`124496f0 0000020a`b33c0c70 0000020b`1516f190 0000020b`124496f0 : System_Private_CoreLib!System.Threading.Tasks.Task.Wait+0x3e
00000049`d11ff6d0 00007ffe`614689fa     : 0000020b`156d5048 0000020a`cfa12c68 0000020a`ab45b388 0000020a`cfa12c68 : Raven_Server!Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler<Raven.Server.ServerWide.Context.DocumentsContextPool,Raven.Server.ServerWide.Context.DocumentsOperationContext>.ReceiveReplicationBatches+0x556
00000049`d11ff9d0 00007ffe`58463d6b     : 0000020b`156d5048 0000020a`b5b75bd8 00000000`00000078 00000000`d11ffa00 : Raven_Server!Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler<Raven.Server.ServerWide.Context.DocumentsContextPool,Raven.Server.ServerWide.Context.DocumentsOperationContext>.DoIncomingReplication+0x1a
00000049`d11ffa20 00007ffe`61127240     : 0000020a`c91a0870 0000020a`abe94bf0 0000020a`ab3c0b00 00007ffe`c9fc3dcb : Raven_Server!Raven.Server.Utils.PoolOfThreads.PooledThread.DoWork+0x3cb
00000049`d11ffb00 00007ffe`565120b3     : 0000020a`c91a0870 0000020a`ad273858 0000020a`ab3c0b00 00000000`00000000 : Raven_Server!Raven.Server.Utils.PoolOfThreads.PooledThread.Run+0xa0
00000049`d11ffbb0 00007ffe`b1ff3073     : 0000020b`0edc6ba0 00000000`00000000 00000000`00000000 0000024b`6f849e00 : System_Private_CoreLib!System.Threading.ExecutionContext.RunInternal+0xe3
```

Tried to do `!dumplog` but it yield `FAILURE: Stress log not dumped`, so I hope all evn variables are set correctly.

Anyway I share the full dump here https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10908179

Please let me know if you require anything more.

**Comment by janvorli** [comment]
@karmeli87 I've investigated the dump and there is clearly a GC hole somewhere. The stress log dump worked fine for me and helped me to reveal some details. There are two objects that the last GC run has logged as having invalid MethodTable and I've found that their references are pointing to the GC heap, but into the middle of another object. So at some point, GC must have scanned the stack and not get their roots reported, then moved objects on the heap and thus these references are pointing wrong places.
Trying to find the real reason for the GC heap is like searching for a needle in a haystack, so it will take me some time. The method in which these two references are stored on the stack is the Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler`2[[System.__Canon, System.Private.CoreLib],[System.__Canon, System.Private.CoreLib]].ReceiveReplicationBatches()
It is huge (almost 8kB) and contains 47 exception handling records (11 original clauses and the rest are cloned and duplicate ones generated by JIT due to specific EH reasons). That makes the investigation even more fun.

**Comment by karmeli87** [comment]
I'm glad to hear that you are able to make some progress on that issue, I know it is fun to find the root cause for such things.
We are able to make it crash pretty often (more or less 1 out of 5 runs of our entire test suite) so, if needed we can share more dumps and add more env variables.

Interesting thing is, that all of the crashes that I've saw crashed at the exact same place.
handling an exception at `Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler`2[[System.__Canon, System.Private.CoreLib],[System.__Canon, System.Private.CoreLib]].ReceiveReplicationBatches()` as you noted.

Which makes use wonder whether it might be related to the exception handling changes that were introduced in .net 9

Please let me know if you need anything that can assist.
Thanks :-)

**Comment by janvorli** [comment]
@karmeli87 it might be related to the EH changes, as the GC stack walk needs to do some special things when exception handling is in progress. Would it be possible for you to try to run with the new EH disabled to see if it gets rid of the crash? You can do it either by setting the env variable DOTNET_LegacyExceptionHandling=1 or by setting System.Runtime.LegacyExceptionHandling to true in the runtimeconfig.json file.

**Comment by janvorli** [comment]
@karmeli87 I've spent another time debugging the dump you've shared with me last time. And I have found something really strange. The crashing thread is in GC preemptive mode, but it should be in cooperative mode at the point where the crash occurs. The preemptive mode causes GC on another thread to walk the stack of this crashing thread freely which should not be happening. I can see that even the GC thread is actually crashing at the same time due to invalid reference it found on the crashing thread's stack.
Did you have a chance to check if the issue still repros with the legacy EH?
I also wonder if you could let your stuff run with a checked build of coreclr.dll that I would provide as an in-place replacement. If that would prove to be too heavy, I could also sprinkle checks for the proper GC mode all over the problematic function and give you a release build.

**Comment by karmeli87** [comment]
@janvorli thanks again for looking into this and sorry for the delayed response.

I've run it many times in EH legacy mode and it didn't crashed, then I switch the legacy mode off to verify and it didn't crashed either (after 10+ tests suite runs I stopped trying).. so it isn't very conclusive.

Of course we can try the custom coreclr.dll 
Should we base it on the latest dotnet?

**Comment by janvorli** [comment]
@karmeli87 I think it would be the best to use the same runtime version that you currently use and just patch it to the checked build.  That way we would be comparing apples to apples. But if the issue doesn't repro anymore, I wonder if some change in the recent runtime servicing updates could have fixed it. What is the current version of the runtime you are using?

**Comment by karmeli87** [comment]
@janvorli we are on `9.0.6`.

This is our development branch so the tests aren't run very often.
When I trigger the test suite manually it might crash 3 times out of 5 runs or not to crash at all for 20 consecutive runs.
But it does crash :-) 

What we will do next is to _always_ run with `DOTNET_LegacyExceptionHandling=1` and observe whether it will crash or not.
Of course we can also use the custom `coreclr.dll`

**Comment by janvorli** [comment]
Ok, let me build and share checked build of coreclr.dll and matching System.Private.CoreLib.dll.

**Comment by janvorli** [comment]
@karmeli87 here is a checked build of clrjit.dll, coreclr.dll, System.Private.CoreLib.dll, mscordaccore.dll and mscordbi.dll and their pdb files built from the v9.0.6 tag. Just copy them over the release files you have.
https://1drv.ms/u/c/2176c86404e3d542/EY9dhXe1CQRBmuohS5t2J9IB8fU-0ACYFOpb7WCDGsf3jA?e=3IUNv7

**Comment by karmeli87** [comment]
@janvorli thanks!
do you prefer us to run this with `DOTNET_LegacyExceptionHandling=1` or set it to `0`

**Comment by janvorli** [comment]
@karmeli87 with it being set to 0.

**Comment by arekpalinski** [comment]
@janvorli It appears that we are no longer able to reproduce the issue. We’re planning to upgrade to .NET 9.0.8. Should we continue using the custom DLLs? If so, could you please provide versions compatible with .NET 9.0.8?

**Comment by janvorli** [comment]
@arekpalinski are you unable to repro it just with the custom dll or even with the original one?

**Comment by arekpalinski** [comment]
It's hard to say since it happened quite occasionally as I know. We upgraded to 9.0.8 today. I’ll monitor for crashes. If any occur, we can install a custom DLL and continue the investigation.

**Comment by janvorli** [comment]
Ok, then you'll need a new build of the custom dlls. They need to match the release. I'll share it later today.

**Comment by arekpalinski** [comment]
@janvorli It just crashed on .NET 9.0.8. Please check the details in the attached file below. Do you want me to upload the crash dump or prefer to wait for it to crash with custom DLL?

[crash_net_9.0.8.txt](https://github.com/user-attachments/files/21715595/crash_net_9.0.8.txt)

**Comment by janvorli** [comment]
Yes, please share the dump. From the stack trace, it seems to be similar, but it is not at the same place.

Here is the custom build for 9.0.8:
https://1drv.ms/u/c/2176c86404e3d542/EazqSsjodxtHsQGKspFe9bIBvG74aLZX-eeLX8t1Mu7b7A?e=UVIOlo

**Comment by arekpalinski** [comment]
@janvorli I have uploaded it to https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10950184

EDIT:

And one more: https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10950200

**Comment by arekpalinski** [comment]
We will install the custom DLLs later today

**Comment by janvorli** [comment]
@arekpalinski thank you for the dump, the issue is the same as before, the only difference being that the crash is in a slightly different method.
In both cases, native runtime helper for virtual stub dispatch stuff is running in GC preemptive mode while that should never happen for that one. That leads to GC running while that thread is modifying managed references on stack and later causes the crash.

**Comment by mangod9** [comment]
We are closing down 10, so moving this to 11. If we find the root cause we can consider back porting.

**Comment by arekpalinski** [comment]
@janvorli I have shared the memory dump of crashing process that had custom .NET DLLs. Console output:

```
Assert failure(PID 5600 [0x000015e0], Thread: 7980 [0x1f2c]): (GetComponentSize() <= 2) || IsArray()
CORECLR! GetCLRRuntimeHost + 0x14C5E5 (0x00007ffb`0f0a8215)
CORECLR! GetCLRRuntimeHost + 0x14F50E (0x00007ffb`0f0ab13e)
CORECLR! GetCLRRuntimeHost + 0x16016E (0x00007ffb`0f0bbd9e)
CORECLR! GetCLRRuntimeHost + 0x160031 (0x00007ffb`0f0bbc61)
CORECLR! GetCLRRuntimeHost + 0x15799D (0x00007ffb`0f0b35cd)
CORECLR! GetCLRRuntimeHost + 0x23259A (0x00007ffb`0f18e1ca)
CORECLR! GetCLRRuntimeHost + 0x68862D (0x00007ffb`0f5e425d)
<no module>! <no symbol> + 0x0 (0x00007ffa`d136c65d)
<no module>! <no symbol> + 0x0 (0x00000225`e9863a08)
<no module>! <no symbol> + 0x0 (0x00000225`e9863a08)
    File: D:\git\runtime9\src\coreclr\vm\methodtable.cpp:6288
```

https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10953395

**Comment by arekpalinski** [comment]
And one more:

```
Assert failure(PID 924 [0x0000039c], Thread: 3396 [0x0d44]): last_gc_info->condemned_generation == max_generation
CORECLR! GetCLRRuntimeHost + 0x648E9C (0x00007fff`82864acc)
CORECLR! GetCLRRuntimeHost + 0x28AE44 (0x00007fff`824a6a74)
<no module>! <no symbol> + 0x0 (0x00007fff`2a5c5a9d)
<no module>! <no symbol> + 0x0 (0x00000000`00000003)
<no module>! <no symbol> + 0x0 (0x00007fff`00000003)
<no module>! <no symbol> + 0x0 (0x000001a0`245c99b0)
<no module>! <no symbol> + 0x0 (0x0000007f`bba7d210)
<no module>! <no symbol> + 0x0 (0x0000007f`bba7dba8)
<no module>! <no symbol> + 0x0 (0x0000007f`bba7ba50)
<no module>! <no symbol> + 0x0 (0x0000007f`bba7d350)
    File: D:\git\runtime9\src\coreclr\gc\gc.cpp:51768
```

https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10953403

**Comment by janvorli** [comment]
@arekpalinski thank you for the dumps! I just got back from my vacation, I'll take a look today or on Monday the latest.

**Comment by janvorli** [comment]
From a quick look, the first dump is related to the culprit, the second one is most likely a different issue. I'll investigate those in more detail. It would be great if you could keep it running for a bit more to get some additional dumps.

**Comment by arekpalinski** [comment]
Yes, we'll keep it running. There is one more dump but it looks the same assertion was encountered as in the second dump that I uploaded:

```
Assert failure(PID 9692 [0x000025dc], Thread: 9444 [0x24e4]): last_gc_info->condemned_generation == max_generation
CORECLR! GetCLRRuntimeHost + 0x648E9C (0x00007ff8`b1034acc)
CORECLR! GetCLRRuntimeHost + 0x28AE44 (0x00007ff8`b0c76a74)
<no module>! <no symbol> + 0x0 (0x00007ff8`55b8ef9d)
<no module>! <no symbol> + 0x0 (0x00000000`00000003)
<no module>! <no symbol> + 0x0 (0x00007ff8`00000003)
<no module>! <no symbol> + 0x0 (0x000002b9`a9f170b0)
<no module>! <no symbol> + 0x0 (0x000000d4`d6afd850)
<no module>! <no symbol> + 0x0 (0x000000d4`d6afe1e8)
<no module>! <no symbol> + 0x0 (0x000000d4`d6afc090)
<no module>! <no symbol> + 0x0 (0x000000d4`d6afd990)
```

**Comment by arekpalinski** [comment]
@janvorli I have uploaded another memory dump that I believe might interest you

https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10957408

```
Assert failure(PID 6160 [0x00001810], Thread: 9596 [0x257c]): !CREATE_CHECK_STRING(!"Detected use of a corrupted OBJECTREF. Possible GC hole.")
CORECLR! coreclr_shutdown_2 + 0xE2DCD (0x00007fff`e260c8cd)
CORECLR! coreclr_shutdown_2 + 0x3A0E0 (0x00007fff`e2563be0)
CORECLR! coreclr_shutdown_2 + 0x3803E (0x00007fff`e2561b3e)
NTDLL! RtlCaptureContext2 + 0x4A6 (0x00007fff`f6de4146)
CORECLR! GetCLRRuntimeHost + 0x16016E (0x00007fff`e1e2bd9e)
CORECLR! GetCLRRuntimeHost + 0x160031 (0x00007fff`e1e2bc61)
CORECLR! GetCLRRuntimeHost + 0x15799D (0x00007fff`e1e235cd)
CORECLR! GetCLRRuntimeHost + 0x23259A (0x00007fff`e1efe1ca)
CORECLR! GetCLRRuntimeHost + 0x68862D (0x00007fff`e235425d)
<no module>! <no symbol> + 0x0 (0x00007fff`a444236a)
    File: D:\git\runtime9\src\coreclr\vm\object.cpp:631
    Image: c:\Jenkins\workspace\Testsv80_Winx64\s\test\SlowTests\bin\Release\net9.0\testhost.exe
```

**Comment by janvorli** [comment]
@arekpalinski thank you for the dump. It again shows exactly the same symptoms. However, I have found that the stress log size I've asked you to set it is insufficient. I have thought that was a maximum possible size, but my records are obviously obsolete. So in the dumps you've shared, some vital logs are missing for figuring out the cuprit.

Could you please change the `DOTNET_TotalStressLogSize=40000000` to 
`DOTNET_TotalStressLogSize=10` ? This small number actually represents a large size, because when it is smaller than 32768, it is multiplied by 1GB. 

I hope this way, I'll finally have all the evidence.

**Comment by arekpalinski** [comment]
@janvorli I see you requested some `DOTNET_xxxxx` env variables to be set (https://github.com/dotnet/runtime/issues/110769#issuecomment-2549605033) but it looks that currently we don't have any for this specific CI build. Can you please provide the complete list of env variables that need to be set there?

**Comment by janvorli** [comment]
@arekpalinski ah, then please set these:
```
DOTNET_StressLog=1 
DOTNET_LogLevel=7 
DOTNET_LogFacility=80103 
DOTNET_StressLogSize=2000000 
DOTNET_TotalStressLogSize=10
```

**Comment by arekpalinski** [comment]
I just uploaded two memory dumps:


- `Assert failure(PID 4260 [0x000010a4], Thread: 4148 [0x1034]): !CREATE_CHECK_STRING(pMT && pMT->Validate())`
https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10958538

- `Assert failure(PID 7424 [0x00001d00], Thread: 12552 [0x3108]): last_gc_info->condemned_generation == max_generation`
https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10958546

Both runs had the environment variables configured as requested above.

**Comment by janvorli** [comment]
Thank you! I'll take a look today.

**Comment by janvorli** [comment]
@arekpalinski I have not realized that the total memory used by the stress log is clipped to 4GB max, so the logs are still missing some records due to too large per-thread buffer specified. Can you please modify the settings  as follows?
```
DOTNET_StressLog=1
DOTNET_LogLevel=7
DOTNET_LogFacility=80103
DOTNET_StressLogSize=800000
DOTNET_TotalStressLogSize=4
```

**Comment by arekpalinski** [comment]
@janvorli just to make absolutely sure, you're asking about changing the following two

```
DOTNET_StressLogSize=2000000 
DOTNET_TotalStressLogSize=10
```

to

```
DOTNET_StressLogSize=800000
DOTNET_TotalStressLogSize=4
```

Is that correct? In particular `DOTNET_TotalStressLogSize` was to 10 and now it will be 4. Please confirm

**Comment by janvorli** [comment]
Right. Btw, the crucial part is setting the `DOTNET_StressLogSize=800000`, this sets the per-thread log buffer size.

**Comment by arekpalinski** [comment]
OK. The env variables have been set accordingly.

**Comment by arekpalinski** [comment]
New memory dump uploaded:

-  `Assert failure(PID 13012 [0x000032d4], Thread: 9264 [0x2430]): last_gc_info->condemned_generation == max_generation`
https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10959501

**Comment by janvorli** [comment]
The "last_gc_info->condemned_generation == max_generation" is a different problem not related to the GC hole issue. I'll route that one to the right folks.

**Comment by arekpalinski** [comment]
OK. I scheduled a couple of test suite runs yesterday evening but non of them failed. I will update once we'll get something else than "last_gc_info->condemned_generation == max_generation"

**Comment by arekpalinski** [comment]
@janvorli would a memory dump with failing assertion "!CREATE_CHECK_STRING(pMT && pMT->Validate())" be interesting to you?

**Comment by janvorli** [comment]
@arekpalinski yes, please.

**Comment by arekpalinski** [comment]
```
Assert failure(PID 10200 [0x000027d8], Thread: 9196 [0x23ec]): !CREATE_CHECK_STRING(pMT && pMT->Validate())
CORECLR! GetCLRRuntimeHost + 0x16018A (0x00007fff`dc8fbdba)
CORECLR! GetCLRRuntimeHost + 0x160031 (0x00007fff`dc8fbc61)
CORECLR! GetCLRRuntimeHost + 0x15799D (0x00007fff`dc8f35cd)
CORECLR! GetCLRRuntimeHost + 0x23259A (0x00007fff`dc9ce1ca)
CORECLR! GetCLRRuntimeHost + 0x68862D (0x00007fff`dce2425d)
<no module>! <no symbol> + 0x0 (0x00007fff`9f86b1a0)
<no module>! <no symbol> + 0x0 (0x00000213`c3839a78)
<no module>! <no symbol> + 0x0 (0x00000213`c3839a78)
CORECLR! coreclr_shutdown_2 + 0x24C7D8 (0x00007fff`dd2462d8)
CORECLR! coreclr_shutdown_2 + 0x24AED0 (0x00007fff`dd2449d0)
```

Uploaded to: https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10961546

**Comment by arekpalinski** [comment]
@janvorli We're considering upgrading the dev branch currently running on .NET 9, where this issue is reproducing, to .NET 10 RC 1. What are your thoughts on this?

**Comment by janvorli** [comment]
@arekpalinski I think it is a good idea. In the last dump you have shared, I've yesterday finally found the evidence of how the problem happens and I am looking into whether the problem is in the JIT generating incorrect GC information or the runtime doing something wrong. So I won't need more dumps from .NET 9 and you moving to .NET 10 would help us find early whether the issue got fixed by some JIT change or if it still persists. We can start with the official build and keep the stresslog settings. If the issue reproes, we could then again move to a custom build of the runtime.

**Comment by arekpalinski** [comment]
@janvorli We got AVE on .NET 10:

```
000000cd`f2438880 00007ff9`b57f5032     : 00000000`00000002 00000259`d152b820 00000000`00000002 00000000`00000180 : KERNELBASE!RaiseFailFastException+0x152
000000cd`f2438e60 00007ff9`b5833054     : 00000000`00000000 000000cd`f2439c80 00000000`00000000 00000000`00000001 : coreclr!WatsonLastChance+0x1b6
000000cd`f2438ec0 00007ff9`b583234b     : 000000cd`f243af00 00007ff9`5f0db748 00000219`324522e0 00000259`c54f0cf0 : coreclr!EEPolicy::LogFatalError+0x65c
000000cd`f2439b80 00007ff9`b57293cc     : 000000cd`f243a2e0 000000cd`f243aa70 000000cd`f243a2e0 000000cd`f2440000 : coreclr!EEPolicy::HandleFatalError+0x133
(Inline Function) --------`--------     : --------`-------- --------`-------- --------`-------- --------`-------- : coreclr!CLRVectoredExceptionHandlerPhase3+0xddbc8
000000cd`f243a180 00007ff9`b564b7c7     : 00000259`d152b820 000000cd`f2440000 000000cd`f2440000 0053002b`002b0033 : coreclr!CLRVectoredExceptionHandlerPhase2+0xddbec
000000cd`f243a1e0 00007ff9`b564b6fd     : 000000cd`f243a2e0 000000cd`f2440000 000000cd`f243a2e0 00000219`585b7ff0 : coreclr!CLRVectoredExceptionHandler+0x8f
000000cd`f243a250 00007ff9`ca85bfc2     : 00007ff9`b564b620 00000000`00000000 00000219`2f770880 00000000`00000000 : coreclr!CLRVectoredExceptionHandlerShim+0xdd
000000cd`f243a2a0 00007ff9`ca811722     : 000000cd`f243aa70 000000cd`f243a580 00000000`00000000 00000000`00000000 : ntdll!RtlpCallVectoredHandlers+0x112
000000cd`f243a340 00007ff9`ca8838ee     : 00000219`58cc17e0 00000219`58cc17e0 0000000c`00000000 00000219`3d318cf8 : ntdll!RtlDispatchException+0x62
000000cd`f243a580 00007ff9`b562661c     : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : ntdll!KiUserExceptionDispatch+0x2e
000000cd`f243ad10 00007ff9`b5628045     : 00000219`2f1ff770 000000cd`f243aef8 000000cd`f243b0a0 0000000c`00000000 : coreclr!VirtualCallStubManager::ResolveWorker+0x44
000000cd`f243aec0 00007ff9`b56af6fd     : 000000cd`f243b100 00000219`58595788 00000219`585b7ff0 00007ff9`66061984 : coreclr!VSD_ResolveWorker+0x335
000000cd`f243aff0 00007ff9`66061a06     : 00000219`5857bff0 00007ff9`6606199f 000000cd`f243c300 000000cd`f243b8c8 : coreclr!ResolveWorkerAsmStub+0x6d
000000cd`f243b0a0 00007ff9`b56ae09a     : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : Raven_Server!Raven.Server.Documents.Replication.Incoming.AbstractIncomingReplicationHandler<Raven.Server.ServerWide.Context.DocumentsContextPool,Raven.Server.ServerWide.Context.DocumentsOperationContext>.ReceiveReplicationBatches+0x2746

```

Full output of `!analyze -v`: [ave_NET_10.txt](https://github.com/user-attachments/files/22507990/ave_NET_10.txt). 

The code in question: https://github.com/ravendb/ravendb/blob/1046a6ebe4416ce3b8b1dca9dce1054a9f2fb367/src/Raven.Server/Documents/Replication/Incoming/AbstractIncomingReplicationHandler.cs#L133

It doesn't seem to be similar to what we saw in the past. Is it something interesting to you? I can provide the memory dump.

**Comment by janvorli** [comment]
@arekpalinski based on the call stack above, this is exactly the same issue we are looking for. In the meanwhile, I was actually able to create a simple repro that triggers the same problem when I make the GC occur at a specific spot in the runtime that I've identified from the stress log in the previous dump you've shared. I am finalizing a fix for the problem.
However, please share the dump so that I can check that the latest failure is not some slightly different failure mode.

**Comment by arekpalinski** [comment]
@janvorli here you go: https://developercommunity.visualstudio.com/t/Access-Violation-Exception-after-upgradi/10814326#T-N10972075

**Comment by janvorli** [comment]
@arekpalinski the fix for this issue was approved for .NET 10 release.

**Comment by wickedmachinator** [comment]
> [@arekpalinski](https://github.com/arekpalinski) the fix for this issue was approved for .NET 10 release.

Thank you very much!

**Comment by arekpalinski** [comment]
> [@arekpalinski](https://github.com/arekpalinski) the fix for this issue was approved for .NET 10 release.

That's awesome. Thank you! Will it be available in any RC before the official .NET 10 release in Nov?

**Comment by janvorli** [comment]
No, it will be in the final release. But I'll let you know once we have an internal build with that fix so that you can test it if you want.

**Comment by janvorli** [comment]
@arekpalinski there is a .NET 10 build with the fix available now: https://github.com/dotnet/dotnet/blob/main/docs/builds-table.md, in the `main (10.0.x Runtime)` column.

**Comment by arekpalinski** [comment]
@janvorli Thank you. Can you help us with finding a NuGet feed with .NET 10 system packages so we'll be able to fix issues like:

```
Unable to find package Microsoft.NETCore.App.Host.osx-x64 with version (= 10.0.0)
        - Found 208 version(s) in api.nuget.org [ Nearest version: 10.0.0-rc.2.25502.107 ]
        - Found 0 version(s) in project
```

**Comment by janvorli** [comment]
This is probably what you need:
https://github.com/dotnet/sdk?tab=readme-ov-file#for-net-10-builds

**Comment by arekpalinski** [comment]
@janvorli we've switched to .NET 10 RTM. I'll provide an update on the results after testing and monitoring the test builds for a couple of days.

**Comment by arekpalinski** [comment]
@janvorli 
After running the RTM build for several days and executing a number of test builds, we can confirm the issue is no longer reproducible on our end. We are closing this issue.

On behalf of the RavenDB team, thank you for your considerable effort in figuring out the root cause and fixing it. We deeply appreciate your hard work; it has been invaluable for us to continue with further product development. Thank you again!

**Comment by janvorli** [comment]
@arekpalinski, @karmeli87 I was happy to help. And thank you for all the help and testing of custom builds during the investigations. Without it, I would probably not be able to identify the real culprit.
